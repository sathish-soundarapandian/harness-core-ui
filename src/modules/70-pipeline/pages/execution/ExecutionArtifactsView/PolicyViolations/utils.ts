import { SortMethod } from '@harness/uicore'
import { useQueryParamsOptions } from '@common/hooks/useQueryParams'

export type TableQueryParams = {
  searchTerm?: string
  page?: number
  size?: number
}

export type EnforcementViolationsParamsWithDefaults = RequiredPick<TableQueryParams, 'page' | 'size' | 'searchTerm'>

export const ENFORCEMENT_VIOLATIONS_PAGE_SIZE = 20
export const ENFORCEMENT_VIOLATIONS_PAGE_INDEX = 0
export const ENFORCEMENT_VIOLATIONS_DEFAULT_SORT: SortMethod = SortMethod.Newest

export const useEnforcementViolationsQueryParamOptions = () => {
  return useQueryParamsOptions({
    page: ENFORCEMENT_VIOLATIONS_PAGE_INDEX,
    size: ENFORCEMENT_VIOLATIONS_PAGE_SIZE,
    sort: ENFORCEMENT_VIOLATIONS_DEFAULT_SORT,
    searchTerm: ''
  })
}
