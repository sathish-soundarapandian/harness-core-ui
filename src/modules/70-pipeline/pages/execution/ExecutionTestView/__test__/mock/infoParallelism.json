[
  {
    "step": "Bucket1_1",
    "stage": "Build_and_Test"
  },
  {
    "step": "Bucket1_0",
    "stage": "Build_and_Test"
  },
  {
    "step": "Bucket1_3",
    "stage": "Build_and_Test"
  },
  {
    "step": "Bucket1_5",
    "stage": "Build_and_Test"
  },
  {
    "step": "Bucket1_4",
    "stage": "Build_and_Test"
  },
  {
    "step": "Bucket1_2",
    "stage": "Build_and_Test"
  }
]
