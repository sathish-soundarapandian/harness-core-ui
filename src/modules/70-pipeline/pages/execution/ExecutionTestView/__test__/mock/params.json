{
  "accountId": "uy6FLHeWTBOOPZA8xchgZA",
  "orgIdentifier": "default",
  "projectIdentifier": "Default_Project_1657212353481",
  "pipelineIdentifier": "mtranjhttp",
  "executionIdentifier": "zRKfRlLUQ1q-W7u6ITyP2g"
}
