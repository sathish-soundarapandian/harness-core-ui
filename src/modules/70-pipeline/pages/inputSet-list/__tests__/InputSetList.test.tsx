/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import {
  render,
  RenderResult,
  waitFor,
  fireEvent,
  getByText,
  queryByText,
  getByTestId,
  screen,
  findByText,
  findByRole
} from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { noop } from 'lodash-es'
import userEvent from '@testing-library/user-event'
import { TestWrapper, findDialogContainer, findPopoverContainer } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import * as pipelineng from 'services/pipeline-ng'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'
import { accountPathProps, pipelineModuleParams, pipelinePathProps } from '@common/utils/routeUtils'
import { branchStatusMock, gitConfigs, sourceCodeManagers } from '@connectors/mocks/mock'
import MonacoEditor from '@common/components/MonacoEditor/__mocks__/MonacoEditor'
import { GetYamlDiffDelResponse } from '@pipeline/components/InputSetErrorHandling/__tests__/InputSetErrorHandlingMocks'
import { PipelineResponse as PipelineDetailsMockResponse } from '../../pipeline-details/__tests__/PipelineDetailsMocks'
import InputSetList from '../InputSetList'
import {
  TemplateResponse,
  PipelineResponse,
  ConnectorResponse,
  GetInputSetsResponse,
  GetInputSetEdit,
  MergeInputSetResponse,
  GetOverlayInputSetEdit,
  GetOverlayISYamlDiff,
  GetInputSetYamlDiff
} from './InputSetListMocks'

const successResponse = (): Promise<{ status: string }> => Promise.resolve({ status: 'SUCCESS', data: {} })
jest.mock('@common/components/YAMLBuilder/YamlBuilder')
jest.mock('react-monaco-editor', () => ({
  MonacoDiffEditor: MonacoEditor
}))

jest.mock('@common/components/MonacoEditor/MonacoEditor')
jest.useFakeTimers()

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),

  useMutateAsGet: jest.fn().mockImplementation(() => {
    return TemplateResponse
  })
}))
const deleteInputSetMock = jest.fn()
const deleteInputSet = (): Promise<{ status: string }> => {
  deleteInputSetMock()
  return Promise.resolve({ status: 'SUCCESS' })
}
const getInputSetList = jest.fn()

jest.mock('@common/components/YAMLBuilder/YamlBuilder')
jest.mock('@common/utils/YamlUtils', () => ({}))
jest.mock('services/cd-ng', () => ({
  useGetConnector: jest.fn(() => ConnectorResponse),
  useCreatePR: jest.fn(() => noop),
  useCreatePRV2: jest.fn(() => noop),
  useGetFileContent: jest.fn(() => noop),
  useGetFileByBranch: jest.fn().mockImplementation(() => ({ refetch: jest.fn() })),
  useGetListOfBranchesWithStatus: jest.fn().mockImplementation(() => {
    return { data: branchStatusMock, refetch: jest.fn(), loading: false }
  }),
  useListGitSync: jest.fn().mockImplementation(() => {
    return { data: gitConfigs, refetch: jest.fn() }
  }),
  useGetSourceCodeManagers: jest.fn().mockImplementation(() => {
    return { data: sourceCodeManagers, refetch: jest.fn() }
  })
}))

jest.mock('services/pipeline-ng', () => ({
  useGetPipeline: jest.fn(() => PipelineResponse),
  useGetPipelineSummary: jest.fn(() => PipelineDetailsMockResponse),
  useGetTemplateFromPipeline: jest.fn(() => TemplateResponse),
  useGetStagesExecutionList: jest.fn(() => ({})),
  useGetMergeInputSetFromPipelineTemplateWithListInput: jest.fn(() => MergeInputSetResponse),
  useGetOverlayInputSetForPipeline: jest.fn(() => GetOverlayInputSetEdit),
  useCreateInputSetForPipeline: jest.fn(() => ({})),
  useUpdateInputSetForPipeline: jest.fn().mockImplementation(() => ({ mutate: successResponse })),
  useUpdateOverlayInputSetForPipeline: jest.fn().mockImplementation(() => ({ mutate: successResponse })),
  useCreateOverlayInputSetForPipeline: jest.fn(() => ({})),
  useGetInputSetsListForPipeline: jest.fn().mockImplementation(args => {
    getInputSetList(args)
    return GetInputSetsResponse
  }),
  getInputSetForPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve(GetInputSetsResponse.data)),
  useGetInputSetForPipeline: jest.fn(() => GetInputSetEdit),
  useDeleteInputSetForPipeline: jest.fn().mockImplementation(() => ({ mutate: deleteInputSet })),
  usePostPipelineExecuteWithInputSetYaml: jest.fn(() => ({})),
  useGetSchemaYaml: jest.fn(() => ({})),
  useSoftDeletePipeline: jest.fn().mockImplementation(() => ({ mutate: jest.fn() })),
  useGetInputsetYaml: jest.fn(() => ({ data: null })),
  useYamlDiffForInputSet: jest.fn(() => ({ data: null }))
}))

const TEST_PATH = routes.toInputSetList({ ...accountPathProps, ...pipelinePathProps, ...pipelineModuleParams })

const intersectionObserverMock = () => ({
  observe: () => null,
  unobserve: () => null
})

window.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock)

const renderComponent = (): RenderResult => {
  return render(
    <TestWrapper
      path={TEST_PATH}
      pathParams={{
        accountId: 'testAcc',
        orgIdentifier: 'testOrg',
        projectIdentifier: 'test',
        pipelineIdentifier: 'pipeline',
        module: 'cd'
      }}
      defaultAppStoreValues={defaultAppStoreValues}
    >
      <InputSetList />
    </TestWrapper>
  )
}

describe('Input Set List tests', () => {
  test('render Input Set List view', async () => {
    const { getAllByText, container } = renderComponent()
    jest.runOnlyPendingTimers()
    await waitFor(() => getAllByText('OverLayInput'))
    expect(container).toMatchSnapshot()
  })
})

describe('Input Set List - Actions tests', () => {
  let container: HTMLElement | undefined
  let getAllByText: RenderResult['getAllByText'] | undefined

  beforeEach(async () => {
    const renderObj = renderComponent()
    container = renderObj.container
    getAllByText = renderObj.getAllByText
    await waitFor(() => getAllByText?.('OverLayInput'))
  })

  test('click handler for Input Set', async () => {
    const inputSetRow = getAllByText?.('asd')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for Overlay Input Set', async () => {
    const inputSetRow = getAllByText?.('OverLayInput')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByText(document.body, 'inputSets.editOverlayTitle'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Overlay Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByText(document.body, 'inputSets.editOverlayTitle'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for delete Overlay Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'delete inputSets.overlayInputSet'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'inputSets.inputSetDeleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for delete Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'delete inputSets.inputSetLabel'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'inputSets.inputSetDeleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for new Overlay Input Set', async () => {
    const menu = getAllByText?.('inputSets.newInputSet')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'inputSets.overlayInputSet')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByText(document.body, 'inputSets.newOverlayInputSet'))
      let form = findDialogContainer()
      expect(form).toBeTruthy()
      // Close
      fireEvent.click(form?.querySelector('[icon="small-cross"]')!)
      form = findDialogContainer()
      expect(form).not.toBeTruthy()
    })
  })

  test('click handler for new Input Set', async () => {
    const menu = getAllByText?.('inputSets.newInputSet')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'inputSets.inputSetLabel')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: '-1',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('search Input Set list', async () => {
    getInputSetList.mockReset()
    const searchInput = container?.querySelector('[placeholder="inputSets.searchInputSet"]') as HTMLInputElement
    fireEvent.change(searchInput, { target: { value: 'asd' } })
    expect(getInputSetList).toBeCalledWith({
      debounce: 300,
      queryParams: {
        accountIdentifier: 'testAcc',
        orgIdentifier: 'testOrg',
        pageIndex: 0,
        pageSize: 20,
        pipelineIdentifier: 'pipeline',
        projectIdentifier: 'test',
        searchTerm: 'asd'
      }
    })
  })
})

describe('Input Set List - Reconcile Button', () => {
  test('should not open reconcile dialog on clicking reconcile button, when loading state is true', async () => {
    jest.spyOn(pipelineng, 'getInputSetForPipelinePromise').mockImplementation((): any => GetInputSetsResponse)
    jest.spyOn(pipelineng, 'useYamlDiffForInputSet').mockImplementation((): any => {
      return {
        data: {
          data: {},
          status: 'SUCCESS'
        },
        loading: true,
        refetch: jest.fn(),
        error: null
      }
    })
    renderComponent()
    await waitFor(() => screen.getAllByText('OverLayInput'))

    const reconcileBtns = await screen.findAllByRole('button', { name: 'pipeline.outOfSyncErrorStrip.reconcile' })
    userEvent.click(reconcileBtns[1])
    expect(pipelineng.useYamlDiffForInputSet).toHaveBeenCalled()

    const reconcileDialog = findDialogContainer() as HTMLElement
    await waitFor(() => expect(reconcileDialog).toBeFalsy())
  })

  test('Input Set - should open reconcile dialog on clicking reconcile button, when loading state is false & input set is not empty', async () => {
    jest.spyOn(pipelineng, 'useYamlDiffForInputSet').mockImplementation((): any => GetInputSetYamlDiff)
    renderComponent()
    await waitFor(() => screen.getAllByText('OverLayInput'))

    const reconcileBtns = await screen.findAllByRole('button', { name: 'pipeline.outOfSyncErrorStrip.reconcile' })
    userEvent.click(reconcileBtns[1])
    expect(pipelineng.useYamlDiffForInputSet).toHaveBeenCalled()

    const reconcileDialog = findDialogContainer() as HTMLElement
    await findByText(reconcileDialog, 'pipeline.inputSetErrorStrip.reconcileDialogTitle')
    const removeInvalidFieldBtn = await findByRole(reconcileDialog, 'button', {
      name: 'pipeline.inputSets.removeInvalidFields'
    })
    userEvent.click(removeInvalidFieldBtn)
    await waitFor(() => expect(pipelineng.useUpdateInputSetForPipeline).toHaveBeenCalled())
  })

  test('Overlay Input Set - should open reconcile dialog on clicking reconcile button, when loading state is false & input set is not empty', async () => {
    jest.spyOn(pipelineng, 'useYamlDiffForInputSet').mockImplementation((): any => GetOverlayISYamlDiff)
    renderComponent()
    jest.runOnlyPendingTimers()
    await waitFor(() => screen.getAllByText('OverLayInput'))

    const reconcileBtns = await screen.findAllByRole('button', { name: 'pipeline.outOfSyncErrorStrip.reconcile' })
    userEvent.click(reconcileBtns[0])
    expect(pipelineng.useYamlDiffForInputSet).toHaveBeenCalled()

    await screen.findAllByText('pipeline.inputSetErrorStrip.reconcileDialogTitle')
    const removeInvalidFieldBtn = await screen.findAllByRole('button', {
      name: 'pipeline.inputSets.removeInvalidFields'
    })
    userEvent.click(removeInvalidFieldBtn[0])
    await waitFor(() => expect(pipelineng.useUpdateOverlayInputSetForPipeline).toHaveBeenCalled())
  })

  test('should open delete input set modal on clicking reconcile button, if input set is empty', async () => {
    jest.spyOn(pipelineng, 'useYamlDiffForInputSet').mockImplementation((): any => GetYamlDiffDelResponse)
    jest.spyOn(pipelineng, 'useDeleteInputSetForPipeline').mockImplementation((): any => {
      return {
        mutate: () =>
          Promise.resolve({
            status: 'SUCCESS'
          })
      }
    })
    renderComponent()
    jest.runOnlyPendingTimers()
    await waitFor(() => screen.getAllByText('OverLayInput'))

    const reconcileBtns = await screen.findAllByRole('button', { name: 'pipeline.outOfSyncErrorStrip.reconcile' })
    userEvent.click(reconcileBtns[0])
    expect(pipelineng.useYamlDiffForInputSet).toHaveBeenCalled()

    const deleteInputSetModal = findDialogContainer() as HTMLElement
    await findByText(deleteInputSetModal, 'pipeline.inputSets.invalidOverlayISDesc2')
    const deleteOverlayISBtn = await findByRole(deleteInputSetModal, 'button', {
      name: 'pipeline.inputSets.deleteOverlayIS'
    })

    userEvent.click(deleteOverlayISBtn)
    await waitFor(() => expect(pipelineng.useDeleteInputSetForPipeline).toHaveBeenCalled())
  })
})
