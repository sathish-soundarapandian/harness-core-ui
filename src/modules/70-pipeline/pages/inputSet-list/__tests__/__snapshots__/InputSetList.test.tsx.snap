// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Set List tests render Input Set List view 1`] = `
<div>
  Help Panel
  <div
    class="Layout--horizontal StyledProps--main PageSubHeader--container StyledProps--flex"
    data-testid="page-subheader"
  >
    <div
      class="Layout--horizontal StyledProps--main"
    >
      <span
        class="bp3-popover-wrapper"
      >
        <span
          class="bp3-popover-target"
        >
          <button
            aria-label="inputSets.newInputSet"
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withRightIcon Button--variation Button--variation-primary"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              inputSets.newInputSet
            </span>
            <span
              class="bp3-icon bp3-icon-caret-down StyledProps--main"
              icon="caret-down"
            >
              <svg
                data-icon="caret-down"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  caret-down
                </desc>
                <path
                  d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </span>
      </span>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
    >
      <div
        class="TextInput--main searchWrapper"
      >
        <div
          class="bp3-input-group"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input"
            placeholder="inputSets.searchInputSet"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody"
    data-name="page-body"
    data-testid="page-body"
  >
    <div
      class="TableV2--table table"
    >
      <div
        class="TableV2--header"
        role="row"
      >
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 30%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            PIPELINE.INPUTSETS.INPUTSETNAMELABEL
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 35%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            DESCRIPTION
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 30%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            ACTION
          </p>
        </div>
        <div
          class="TableV2--cell"
          colspan="1"
          role="columnheader"
          style="width: 5%;"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
          >
            
          </p>
        </div>
      </div>
      <div
        class="TableV2--body"
      >
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 30%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-blue500"
                  data-icon="step-group"
                >
                  <svg
                    height="30"
                    id="step-group_svg__Layer_1"
                    viewBox="0 0 21 20"
                    width="30"
                    x="0"
                    xml:space="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    y="0"
                  >
                    <style>
                      .step-group_svg__st0{fill:#fff!important}
                    </style>
                    <path
                      class="step-group_svg__st0"
                      d="M1 .8h13v13.3H1z"
                    />
                    <path
                      d="M14 1H1v13h13V1zm1-1v15H0V0h15z"
                    />
                    <path
                      class="step-group_svg__st0"
                      d="M7 6h13v13H7z"
                    />
                    <path
                      class="step-group_svg__st1"
                      d="M20 6H7v13h13V6zm1-1v15H6V5h15z"
                    />
                  </svg>
                </span>
                <div
                  class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--padding-right-medium"
                  style="flex-shrink: 1;"
                >
                  <div>
                    <div
                      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      data-testid="OverLayInput"
                    >
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
                        style="--text-line-clamp: 1;"
                      >
                        OverLayInput
                      </p>
                    </div>
                    <p
                      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
                      style="--text-line-clamp: 1;"
                    >
                      idLabel
                    </p>
                  </div>
                  <div
                    class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
                  >
                    <span
                      class="bp3-icon StyledProps--main badgeIcon StyledProps--intent-danger"
                      data-icon="error-outline"
                      data-testid="invalid-icon"
                    >
                      <svg
                        fill="none"
                        height="20"
                        viewBox="0 0 16 16"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.423 1.631a.667.667 0 011.154 0l6.543 11.333a.667.667 0 01-.577 1H1.457a.667.667 0 01-.577-1L7.423 1.631z"
                          fill="#fff"
                        />
                        <path
                          clip-rule="evenodd"
                          d="M7.57 2.253a.5.5 0 01.869 0l5.993 10.5a.5.5 0 01-.434.747H2.011a.5.5 0 01-.434-.748L7.57 2.253zm.407 8.242c.422 0 .763.333.763.744a.754.754 0 01-.763.745.754.754 0 01-.763-.745c0-.411.341-.745.763-.745zm.5-1.498a.5.5 0 11-1 0V5.498a.5.5 0 111 0v3.499z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <p
                class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--padding-right-medium StyledProps--color StyledProps--color-black"
                style="--text-line-clamp: 2;"
              >
                OverLayInput
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 30%;"
            >
              <button
                aria-label="pipeline.outOfSyncErrorStrip.reconcile"
                class="bp3-button Button--button StyledProps--font StyledProps--main reconcileButtonListView Button--with-current-color Button--variation Button--variation-secondary Button--size Button--small"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  pipeline.outOfSyncErrorStrip.reconcile
                </span>
              </button>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <div
                class="Layout--horizontal StyledProps--main"
                style="justify-content: flex-end;"
              >
                <span
                  class="bp3-popover-wrapper bp3-dark"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <button
                      class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main actionButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-more StyledProps--main"
                        icon="more"
                      >
                        <svg
                          data-icon="more"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            more
                          </desc>
                          <path
                            d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="TableV2--row TableV2--card TableV2--clickable"
          role="row"
        >
          <div
            class="TableV2--cells"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 30%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-black"
                  data-icon="yaml-builder-input-sets"
                >
                  <svg
                    height="30"
                    viewBox="0 0 18 16"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#22272D"
                    >
                      <path
                        d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                      />
                      <path
                        d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                      />
                    </g>
                  </svg>
                </span>
                <div
                  class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--padding-right-medium"
                  style="flex-shrink: 1;"
                >
                  <div>
                    <div
                      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                      data-testid="asd"
                    >
                      <p
                        class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-black"
                        style="--text-line-clamp: 1;"
                      >
                        asd
                      </p>
                    </div>
                    <p
                      class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
                      style="--text-line-clamp: 1;"
                    >
                      idLabel
                    </p>
                  </div>
                  <div
                    class="StyledProps--font StyledProps--main StyledProps--padding-left-large"
                  >
                    <span
                      class="bp3-icon StyledProps--main badgeIcon StyledProps--intent-danger"
                      data-icon="error-outline"
                      data-testid="invalid-icon"
                    >
                      <svg
                        fill="none"
                        height="20"
                        viewBox="0 0 16 16"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M7.423 1.631a.667.667 0 011.154 0l6.543 11.333a.667.667 0 01-.577 1H1.457a.667.667 0 01-.577-1L7.423 1.631z"
                          fill="#fff"
                        />
                        <path
                          clip-rule="evenodd"
                          d="M7.57 2.253a.5.5 0 01.869 0l5.993 10.5a.5.5 0 01-.434.747H2.011a.5.5 0 01-.434-.748L7.57 2.253zm.407 8.242c.422 0 .763.333.763.744a.754.754 0 01-.763.745.754.754 0 01-.763-.745c0-.411.341-.745.763-.745zm.5-1.498a.5.5 0 11-1 0V5.498a.5.5 0 111 0v3.499z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <p
                class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--padding-right-medium StyledProps--color StyledProps--color-black"
                style="--text-line-clamp: 2;"
              >
                asd
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 30%;"
            >
              <button
                aria-label="pipeline.outOfSyncErrorStrip.reconcile"
                class="bp3-button Button--button StyledProps--font StyledProps--main reconcileButtonListView Button--with-current-color Button--variation Button--variation-secondary Button--size Button--small"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  pipeline.outOfSyncErrorStrip.reconcile
                </span>
              </button>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <div
                class="Layout--horizontal StyledProps--main"
                style="justify-content: flex-end;"
              >
                <span
                  class="bp3-popover-wrapper bp3-dark"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <button
                      class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main actionButton Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-more StyledProps--main"
                        icon="more"
                      >
                        <svg
                          data-icon="more"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            more
                          </desc>
                          <path
                            d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
